lib/builderator/util.rb

```diff
@@ -60,8 +60,15 @@ def filter!(resources, filters = {})
       ##
       # AWS Clients
       ##
-      def ec2(region = Config.aws.region)
-        clients["ec2-#{region}"] ||= Aws::EC2::Client.new(:region => region)
+      def ec2(region = Config.aws.region, credentials=nil)
+        options = { :region => region }
+
+        # Don't memoize if supplying explicit credentials as it could be an assumed role for a remote account
+        if credentials.nil?
+          clients["ec2-#{region}"] ||= Aws::EC2::Client.new options
+        else
+          Aws::EC2::Client.new options.merge(credentials)
```

Comment:
I don't think so...depending on whether I understand what you mean. The `options` hash should contain `{ :region => x, :access_key_id => y}`, etc.  It's a flat hash without nested hashes.